File: api/KS.FiksProtokollValidator.WebAPI/Models/TestCase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace KS.FiksProtokollValidator.WebAPI.Models
{
    public class TestCase
    {
        [Key]
        public string TestName { get; set; }

        [Required]
        public string MessageType { get; set; }

        [Required]
        public string PayloadFileName { get; set; }

        public string PayloadAttachmentFileNames { get; set; }

        public List<FiksResponseTest> FiksResponseTests { get; set; }

        [Required]
        public string Description { get; set; }
        [Required]
        public string TestStep { get; set; }
        [Required]
        public string Operation { get; set; }
        [Required]
        public string Situation { get; set; }
        [Required]
        public string ExpectedResult { get; set; }
        [Required]
        public bool Supported { get; set; }
        [Required]
        public List<FiksExpectedResponseMessageType> ExpectedResponseMessageTypes { get; set; }
    }
}

File: web-ui/src/types/testCase.ts
export type ValidationState = 'valid' | 'invalid' | 'notValidated'

export interface TestCase {
  testId: string
  testName: string
  messageType: string
  description: string
  testStep: string
  operation: string
  situation: string
  expectedResult: string
  payloadFileName?: string
  payloadAttachmentFileNames?: string
  supported: boolean
  protocol: string
  fiksResponseTests?: FiksResponseTest[]
}

export interface FiksResponseTest {
  id: string
  payloadQuery: string
  valueType: number
  expectedValue: string
}

export interface FiksPayload {
  id: string
  filename: string
  payload?: string
}

export interface FiksResponse {
  id: string
  receivedAt: string
  type: string
  fiksPayloads: FiksPayload[]
  payloadContent?: string
}

export interface FiksRequest {
  messageGuid: string
  sentAt: string
  testCase: TestCase
  fiksResponses: FiksResponse[]
  customPayloadFile?: {
    filename: string
    content?: string
  }
  isFiksResponseValidated: boolean
  fiksResponseValidationErrors: string[]
}

export interface TestSession {
  id: string
  recipientId: string
  protocol: string
  status: string
  createdAt: string
  completedAt?: string
  fiksRequests: FiksRequest[]
}

export interface ProtocolOption {
  value: string
  text: string
}
